src/Host/Broker/Impl/NativeMethods.cs

```diff
@@ -52,5 +94,19 @@ internal static unsafe class NativeMethods {
             int domainMaxChars);
         [DllImport("userenv.dll", SetLastError = true, CharSet = CharSet.Auto)]
         public static extern bool GetUserProfileDirectory(IntPtr hToken, StringBuilder pszProfilePath, ref uint cchProfilePath);
+
+        [DllImport("kernel32.dll")]
+        static extern ulong VerSetConditionMask(ulong dwlConditionMask, uint dwTypeBitMask, byte dwConditionMask);
+
+        [DllImport("kernel32.dll")]
+        static extern bool VerifyVersionInfo([In] ref OSVERSIONINFOEX lpVersionInfo, uint dwTypeMask, ulong dwlConditionMask);
+
+        public static bool IsWindowsServer() {
+            OSVERSIONINFOEX osVersionInfo = default(OSVERSIONINFOEX);
+            osVersionInfo.dwOSVersionInfoSize = (uint)Marshal.SizeOf(osVersionInfo);
+            osVersionInfo.wProductType = VER_NT_WORKSTATION;
+            ulong mask = VerSetConditionMask(0, VER_PRODUCT_TYPE, VER_EQUAL);
```

Comment:
https://msdn.microsoft.com/en-us/library/bb773795(v=VS.85).aspx may be simpler to use